src/ops/ops.go

```diff
@@ -77,3 +83,33 @@ func (o *ops) Reboot() error {
 	}
 	return nil
 }
+
+func (o *ops) AddPullSecretToDockerConfig(ignitionPath string) error {
```

Comment:
You can make it generic
```suggestion
func (o *ops) extractFromIgnition(ignitionPath string, targetPath string) error {
```

Reply:
It can't be generic, there is no point for it. Generic part of ignition parsing is in utils.
This func read file , parse in util func, write it to tmp path and move it to docker config